website/docs/stitch-directives-sdl.md

```diff
@@ -0,0 +1,348 @@
+
+Stitching directives (`@graphql-tools/stitching-directives`) may be used to configure a stitched gateway schema directly through the Schema Definition Language (SDL, also referred to as "type definitions") of its subservices. The advantage of this approach is that all schema _and stitching configuration_ is represented in a single document managed by each subservice, and can be loaded (or reloaded) by the gateway on the fly. These SDL configurations enable hot-reloading of the gateway schema without a formal deployment or server restart.
+
+## Overview
+
+Using SDL directives, a subservice may express its complete schema _and type merging configuration_ in a single document, which may then be pushed to the gateway server to trigger a reload.
+
+```graphql
+# --- Users schema ---
+type User @key(selectionSet: "{ id }") {
+  id: ID!
+  username: String!
+  email: String!
+}
+
+type Query {
+  users(ids: [ID!]!): [User]! @merge(keyField: "id")
+}
+
+# --- Posts schema ---
+type Post {
+  id: ID!
+  message: String!
+  author: User
+}
+
+type User @key(selectionSet: "{ id }") {
+  id: ID!
+  posts: [Post]
+}
+
+type Query {
+  post(id: ID!): Post
+  _users(ids: [ID!]!): [User]! @merge(keyField: "id")
+}
+```
+
+In the above example, the Users and Posts schemas will be combined in the stitched gateway and provide all of their own merged type configuration. All SDL directives will translate directly into the static configurations discussed in [type merging docs](/docs/stitch-type-merging). See the [recipes](#recipes) section below for some common translations.
+
+## Directives glossary
+
+By default, stitching directives use the following definitions (though the names of these directives [may be customized](#customizing-names)):
+
+```graphql
+directive @key(selectionSet: String!) on OBJECT
+directive @merge(keyField: String, keyArg: String, key: [String!], argsExpr: String, additionalArgs: String) on FIELD_DEFINITION
+directive @computed(selectionSet: String!) on FIELD_DEFINITION
+```
+
+The function of these directives are:
+
+* **`@key`:** specifies the base selection set used to merge the annotated type across subschemas. Analogous to the `selectionSet` setting specified in [merged type configuration](/docs/stitch-type-merging#basic-example).
+
+* **`@merge`:** denotes a root field used to merge a type across services. The marked field's name is analogous to the `fieldName` setting in merged type configuration, and the field's arguments and return type provide details used in setting up the merger. Additional arguments may tune the merge behavior:
+
+  * `keyField`: specifies the name of a field to pick off the original object as the key value. Generally matches the field specified in the type-level selection set.
+  * `keyArg`: specifies which field argument receives the merge key. This may be omitted for fields with only one argument, thus allowing the argument to be inferred.
+  * `key`: tktk
+  * `argsExpr`: tktk
+  * `additionalArgs`: tktk
```

Comment:
@yaacovCR – would you mind writing some notes in for these three? I'm not familiar with them.

Reply:
https://github.com/ardatan/graphql-tools/blob/master/packages/stitching-directives/tests/stitchingDirectivesTransformer.test.ts

Has a bunch of examples.

Key by default has shape of selectionSet, but you can use key argument to override, array of strings to select from selectionSet using dot notation, you can place a field from selectionSet anywhere in arg by using an alias, also with dot notation, if the shape of the Resolvers argument differs from shape of selectionSet. additionalArgs is for specifying a bunch of additional arguments that the resolver might need, like scope or whatever, argsExpr let's you forgo all the other arguments and write a single expression that fits the same format, see that file for examples, generally no need to use, but available and in implementation, everything gets parsed into that...

Reply:
One more question then: is the name “_Key” significant in any way for the key scalar? You said merge is doing all the lifting and the scalar just receives, which makes me now think the name of it isn’t significant like I initially thought.

Reply:
No significance whatsoever.